src/lang/lang.ml

```diff
@@ -771,7 +771,7 @@ let source_methods =
     ( "self_sync",
       ([], fun_t [] bool_t),
       "Is the source currently controling its own real-time loop.",
-      fun s -> val_fun [] (fun _ -> bool s#self_sync) );
+      fun s -> val_fun [] (fun _ -> bool (snd s#self_sync)) );
```

Comment:
Maybe would it be better to add two fields instead of having `s#self_sync` a pair, which forces us using `snd`?

Reply:
Two methods? That was my original approach but it would require to modify every single operator. Lot of changes, chances for mistakes while the inference logic is actually exactly the same for 90% of them..